### PR TITLE
Add Edge Zone SIG definition

### PR DESCRIPTION
**Pull request by yunrugemicrosoft**
**What type of PR is this?**
/feature

**What this PR does / why we need it**:
Add Edge Zone SIG definition. Update consts with the image gallery name and resource group that stores Edge Zone SIG images.

**Which issue(s) this PR fixes**:

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/sig_config.go [hunk 1]
```diff
@@ -223,13 +228,15 @@ var AvailableWindowsPIRDistros []Distro = []Distro{
...
-	AKSCBLMarinerGalleryName   string = "AKSCBLMariner"
-	AKSCBLMarinerResourceGroup string = "AKS-CBLMariner"
+	AKSSIGImagePublisher           string = "microsoft-aks"
+	AKSWindowsGalleryName          string = "AKSWindows"
+	AKSWindowsResourceGroup        string = "AKS-Windows"
+	AKSUbuntuGalleryName           string = "AKSUbuntu"
+	AKSUbuntuResourceGroup         string = "AKS-Ubuntu"
+	AKSCBLMarinerGalleryName       string = "AKSCBLMariner"
+	AKSCBLMarinerResourceGroup     string = "AKS-CBLMariner"
+	AKSUbuntuEdgeZoneGalleryName   string = "AKSUbuntu_edgezone"
```
Comment (alexeldeib) [inline pkg/agent/datamodel/sig_config.go:238]: Please remove the underscore - https://learn.microsoft.com/en-us/azure/virtual-machines/create-gallery?tabs=portal%2Cportaldirect%2Ccli2#naming

```
Allowed characters for gallery name are uppercase or lowercase letters, digits, dots, and periods. The gallery name can't contain dashes. Gallery names must be unique within your subscription.
```

**Comment by yunrugemicrosoft** [comment]
@microsoft-github-policy-service agree [company="microsoft"]

File: pkg/agent/datamodel/sig_config.go [hunk 2]
```diff
@@ -439,25 +462,27 @@ var (
...
+		AKSUbuntuGPUContainerd1804:          SIGUbuntuGPUContainerd1804ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuGPUContainerd1804Gen2:      SIGUbuntuGPUContainerd1804Gen2ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuFipsContainerd1804:         SIGUbuntuFipsContainerd1804ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuFipsContainerd1804Gen2:     SIGUbuntuFipsContainerd1804Gen2ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuFipsGPUContainerd1804:      SIGUbuntuFipsGPUContainerd1804ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuFipsGPUContainerd1804Gen2:  SIGUbuntuFipsGPUContainerd1804Gen2ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuArm64Containerd1804Gen2:    SIGUbuntuArm64Containerd1804Gen2ImageConfigTemplate.WithOptions(opts...),
+		AKSUbuntuEdgeZoneContainerd1804:     SIGUbuntuEdgeZoneContainerd1804ImageConfigTemplate.WithOptions(),     // Not a typo, prevents the image config value overrided with options.
+		AKSUbuntuEdgeZoneContainerd1804Gen2: SIGUbuntuEdgeZoneContainerd1804Gen2ImageConfigTemplate.WithOptions(), // Not a typo, prevents the image config value overrided with options.
```
Comment (alexeldeib) [inline pkg/agent/datamodel/sig_config.go:480]: what is the meaning of this comment? why not take the opts?

**Comment by alexeldeib** [comment]
please also update GetSIGAzureCloudSpecConfig  if the RG/gallery name changes

**Review by alexeldeib — APPROVED** [review]

**Pull request merged**